src/components/tree.js

```diff
@@ -1,185 +1,227 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import '../App.css';
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import "../App.css";
 
-class Tree extends Component {
+const Tree = (props) => {
+  const [state, setState] = useState({ treeData: [] });
 
-  constructor(props){
-    super(props);
-    this.state = {
-      treeData: []
+  useEffect(() => {
+    const { data } = props;
+    if (data) {
+      setState({ ...state, treeData: treeInitialization(data) });
```

Comment:
don't call function in state update. write like this;
const treeData = treeInitialization(data);
setState(prevState => return { ...prevState, treeData });